app/scripts/proactive/templates/job-variable-template.html

```diff
@@ -118,7 +118,15 @@ <h4 class="modal-title"><%- jobName %> </h4>
             <% } else if (model && model.toLowerCase().indexOf('pa:credential')!=-1) {  %>
                 <textarea class="variableValue textareavalues submitVariableValue" rows=1 name="<%- variable.Name %>" data-variable-model="<%- model %>" id="<%- key %>" ><%- variable.Value %></textarea>
                 <label class="short" data-toggle="tooltip" title="<%- model %>"><%- model ? ((model.length >= 20) ? model.substring(0, 20) + " ..." : model):""  %></label>
-                <button id="third-party-credential-button" type="button" class="btn" style="background-color:Transparent"><img src="images/key.png"></button>
+                <button title="Manage third-party credentials" class="third-party-credential-button btn" type="button" style="background-color:Transparent"><img src="images/key.png"></button>
+            <% } else if (model && model.toLowerCase().indexOf('pa:global_file')!=-1) {  %>
+                <textarea class="variableValue textareavalues submitVariableValue" rows=1 name="<%- variable.Name %>" data-variable-model="<%- model %>" id="<%- key %>" ><%- variable.Value %></textarea>
+                <label class="short" data-toggle="tooltip" title="<%- model %>"><%- model ? ((model.length >= 20) ? model.substring(0, 20) + " ..." : model):""  %></label>
+                <button title="Browse global space files" value="<%- key %>" class="var-globalfile-button btn far fa-folder-open" type="button" style="background-color:Transparent"></button>
+            <% } else if (model && model.toLowerCase().indexOf('pa:user_file')!=-1) {  %>
+                <textarea class="variableValue textareavalues submitVariableValue" rows=1 name="<%- variable.Name %>" data-variable-model="<%- model %>" id="<%- key %>" ><%- variable.Value %></textarea>
+                <label class="short" data-toggle="tooltip" title="<%- model %>"><%- model ? ((model.length >= 20) ? model.substring(0, 20) + " ..." : model):""  %></label>
```

Comment:
Can't you create a method for `model ? ((model.length >= 20) ? model.substring(0, 20) + " ..." : model):""` instead of duplicate it ?

Reply:
Actually I found that it's only necessary for the models like DATETIME which could be too long, so I'm removing those who don't need this expression.